src/hooks/useWatchPendingTxs.ts

```diff
@@ -16,8 +16,12 @@ import { useNonceStore } from '@/state/nonces';
 export const useWatchPendingTransactions = ({ address }: { address: string }) => {
   //const { swapRefreshAssets } = useSwapRefreshAssets();
 
-  const { pendingTransactions: storePendingTransactions, setPendingTransactions } = usePendingTransactionsStore();
-  const { setNonce } = useNonceStore();
+  const { storePendingTransactions, setPendingTransactions } = usePendingTransactionsStore(state => ({
+    storePendingTransactions: state.pendingTransactions,
+    setPendingTransactions: state.setPendingTransactions,
+  }));
```

Comment:
I'm pretty sure this still rerenders every time, since it's creating a new object
iirc it is recommended in the docs to do it like this
```ts
 const storePendingTransactions = usePendingTransactionsStore(state => state.pendingTransactions);
 const setPendingTransactions = usePendingTransactionsStore(state => state.setPendingTransactions);
```